### PR TITLE
Fix typo

### LINKED ISSUE #6977: Possible typo
In the Essential Samples section at the union part, the GetHashCode(Dog d) function shouldn't return b's but d's GetHashCode() function.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 89e7bc95-4062-e6df-7475-21973ed1becc
* Version Independent ID: 571fd6fd-95b6-6fc9-d4df-9b44da6a01b4
* Content: [LINQ (Language Integrated Query)](https://docs.microsoft.com/en-us/dotnet/standard/using-linq)
* Content Source: [docs/standard/using-linq.md](https://github.com/dotnet/docs/blob/master/docs/standard/using-linq.md)
* Product: **dotnet**
* GitHub Login: @cartermp
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by kunklir**
## Summary

Fixed the GetHashCode(Dog d) function at the union part. It now calls d object's function instead of b's.

Fixes #6977

### CODE REVIEW

File: docs/standard/using-linq.md [hunk 1]
```diff
@@ -168,7 +168,7 @@ public class DogHairLengthComparer : IEqualityComparer<Dog>
     public int GetHashCode(Dog d)
     {
         // default hashcode is enough here, as these are simple objects.
-        return b.GetHashCode();
+        return d.GetHashCode();
```
Comment (svick) [inline docs/standard/using-linq.md]: I think this code is still wrong, but I also think that's outside the scope of this Pull Request, so I opened a separate issue about that: https://github.com/dotnet/docs/issues/7002.

**Review by briandunnington — APPROVED** [review]
Indeed, it does fix the typo

File: docs/standard/using-linq.md (hunk 1 above)
Comment (mairaw) [reply docs/standard/using-linq.md]: Yeah, let's keep this one focused on the typo

**Review by mairaw — APPROVED** [review]
Thanks @kunkliricsi for submitting the fix. Your fix should be live within a day.

**Pull request merged**